src/Simulation/Simulators/SparseSimulator/Native/SparseSimulator.h

```diff
@@ -0,0 +1,846 @@
+#pragma once
+#include <string>
+#include "types.h"
+#include "gates.h"
+#include <random>
+#include <cmath>
+#include <functional>
+#include "quantum_state.hpp"
+#include "basic_quantum_state.hpp"
+#include <algorithm>
+#include <iomanip>
+#include <iostream>
+#include <list>
+
+#include <set>
+
+
+#define MAX_QUBITS 1024
+#define MIN_QUBITS 64
+
+using namespace std::literals::complex_literals;
+
+namespace Microsoft
+{
+namespace Quantum
+{
+namespace SPARSESIMULATOR
+{
```

Comment:
Since C++17 is enabled, you should be able to collapse these definitions.
```suggestion
namespace Microsoft::Quantum::SPARSESIMULATOR
{
```

Reply:
Addressed here: https://github.com/sam-jaques/qsharp-runtime/pull/1